### PR TITLE
feat/v1.5.1

### PR DESCRIPTION
**Pull request by nokchalatte**
# Description (개요) 

- Updated docs for post query to integrate query template
- New docs for query template CRUD API
- Changed logic for table template API

## Type of change (작업사항)

- [x] REST API docs update

Following templates can be found from the [Tech-wiki Docs](https://github.com/smartmind-team/tech-wiki/tree/main/project/thanosql/docs). 

## Writing Convention

Make sure to check out the [Writing Convention](https://github.com/smartmind-team/tech-wiki/blob/main/project/thanosql/docs/writing_convention.md). 

## Review Process

Review should be conducted by at least two people.

### CODE REVIEW

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 1]
```diff
@@ -16,23 +16,33 @@ Execute ThanoSQL queries and receive a query log as a response.
...
-    r = requests.post(api_url, data=json.dumps(data), headers=header)
+    # For query with templates (pick 1 between this and above):
+    template_name = "Name of template to use"
+    parameters = "Mapping of parameters to fill in the template"
 
-    r.raise_for_status()
-    return_json = r.json()
+    data = {
+        'query_type': query_type,
+        'template_name': template_name,  # or 'template_id'
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]: I think we need to be more verbose with our description regarding template_id

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 2]
```diff
@@ -94,10 +105,55 @@ This method retrieves a paginated list of all query logs.
...
+    ```shell
+      curl -X 'GET' \
+      'https://{your-engine-url}/api/v1/query/log/?search={search}&offset={offset}&limit={limit}' \
+      -H 'accept: application/json' \
+      -H 'Authorization: Bearer Issued_API_TOKEN'
+    ```
+
+### __Parameters__
+
+- `search`: Word(s) that the returned query logs should contain.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
- `search`: Word(s) that the query in the query log should contain.
```
We are searching for words in the query part of the query log so change the description to something in the same grain as above

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 3]
```diff
@@ -121,3 +177,188 @@ This method retrieves a paginated list of all query logs.
...
+      -H 'accept: application/json' \
+      -H 'Authorization: Bearer Issued_API_TOKEN' \
+      -H 'Content-Type: application/json' \
+      -d '{"name": name, "query": query}'
+    ```
+
+
+## **`GET` /query/template/{template_name}**
+
+Retrieves the query template with a certain name.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
Retrieves the query template with the given name.
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 4]
```diff
@@ -121,3 +177,188 @@ This method retrieves a paginated list of all query logs.
...
+    curl -X 'GET' \
+      'https://{your-engine-url}/api/v1/query/template/{template_name}' \
+      -H 'accept: application/json' \
+      -H 'Authorization: Bearer Issued_API_TOKEN'
+    ```
+
+
+## **`PUT` /query/template/{template_name}**
+
+Updates a query template with a certain name.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
Updates the query template with the given name.
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 5]
```diff
@@ -121,3 +177,188 @@ This method retrieves a paginated list of all query logs.
...
+      -d '{"name": name, "query": query}'
+    ```
+
+!!! Note
+    If you only want to update one field, leave the other field out of the data JSON. For example, if you only want to update the name of the query template, use `{"query": query}` as the data body.
+
+
+## **`DELETE` /query/template/{template_name}**
+
+Deletes a query template with a certain name.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
Deletes the query template with the given name.
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md [hunk 6]
```diff
@@ -42,15 +47,21 @@ In order to get a list of all of your table templates, use the method below.
...
       curl -X 'GET' \
-      'https://{your-engine-url}/api/v1/table_template/' \
+      'https://{your-engine-url}/api/v1/table_template/?search={search}&order_by={order_by}&latest={latest}' \
       -H 'accept: application/json' \
       -H 'Authorization: Bearer Issued_API_TOKEN'
     ```
 
+### __Parameters__
+
+- `search`: Word(s) that the names of queried table templates must contain (defaults to empty string).
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md]:
```suggestion
- `search`: Word(s) that the table templates' name must contain (defaults to empty string).
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md [hunk 7]
```diff
@@ -42,15 +47,21 @@ In order to get a list of all of your table templates, use the method below.
...
+- `order_by`: How the query results should be ordered, there are three possible values: 'recent', based on date of creation from latest to oldest, 'name_asc', based on name from A to Z, and 'name_desc', based on name from Z to A (defaults to 'recent').
+- `latest`: A boolean value that can be set to `True` if you only want to return the latest version of each table template. Otherwise, it returns all available versions (default: `False`).
+
 
 ## **`GET` /table_template/{table_template_name}**
 
-Use this method to get the most recent version of table template object of a certain name.
+Use this method to get the table template object of a certain name. If needed, a specific or latest version can also be added to the request.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md]:
```suggestion
Use this method to get the table template object with the given name. If needed, a specific or latest version can also be added to the request.
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md [hunk 8]
```diff
@@ -77,11 +89,22 @@ Use this method to get the most recent version of table template object of a cer
...
       curl -X 'GET' \
-      'https://{your-engine-url}/api/v1/table_template/{table_template_name}' \
+      'https://{your-engine-url}/api/v1/table_template/{table_template_name}?version={version}' \
       -H 'accept: application/json' \
       -H 'Authorization: Bearer Issued_API_TOKEN'
     ```
 
+### __Response__
+
+This API's response has the following structure:
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md]:
```suggestion
The API response has the following structure:
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md [hunk 9]
```diff
@@ -212,3 +234,39 @@ Use this method to create a new table template.
         "compatibility": "string"
         }'
     ```
+
+
+## **`DELETE` /table_template/{table_template_name}**
+
+Use this method to delete table templates of a certain name. If needed, it is also possible to delete only a specific version.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md]:
```suggestion
Use this method to delete table templates with the given name. If needed, it is also possible to delete only a specific version.
```

**Review by stevensjcho — COMMENTED** [review]
Reviewed the english part of the docs. I will review the Korean one after the english ones comments are addressed first. Let's make sure that the overall tone matches previous documentations

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 10]
```diff
@@ -6,44 +6,60 @@ title: Query APIs
...
 
     ```python
     import requests
+    import json
 
-    api_token = "발급받은_API_TOKEN"
-    api_url="https://{your-engine-url}/api/v1/query/"
-    query="요청할 쿼리"
-    query_type="쿼리를 실행할 sql type" #psql or thanosql
+    api_token = "Issued_API_TOKEN"
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
    api_token = "발급받은_API_TOKEN"
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 11]
```diff
@@ -6,44 +6,60 @@ title: Query APIs
...
+    api_url = "https://{your-engine-url}/api/v1/query/"
+    query_type = "SQL query type" #psql 또는 thanosql
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
    query_type = "쿼리를 실행할 sql type" #psql 또는 thanosql
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 12]
```diff
@@ -6,44 +6,60 @@ title: Query APIs
...
 
     header = {
         "Authorization": f"Bearer {api_token}"
     }
 
+    # 템플릿 없는 쿼리의 경우(아래와 중 하나 선택합니다):
+    query = "Query to request"
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
    query = "요청할 쿼리"
```

**Review by stevensjcho — COMMENTED** [review]
Please match the formatting of other Korean documentations. Especially values that are in between ""

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 13]
```diff
@@ -6,44 +6,60 @@ title: Query APIs
...
 
 === "cURL"
 
-    ```shell 
+    ```shell
     curl -X 'POST' \
       'https://{your-engine-url}/api/v1/query/' \
       -H 'accept: application/json' \
-      -H 'Authorization: Bearer 발급받은_API_TOKEN' \
+      -H 'Authorization: Bearer Issued_API_TOKEN' \
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
      -H 'Authorization: Bearer 발급받은_API_TOKEN' \
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 14]
```diff
@@ -80,20 +96,65 @@ ThanoSQL 쿼리를 실행하고 쿼리 로그를 응답으로 받습니다.
...
+### __Parameters__
+
+- `search`: 반환된 쿼리 로그에 포함되어야 하는 단어입니다.
+- `offset`: pagination count이 시작되는 오프셋입니다(기본값은 0).
+- `limit`: 오프셋에서 시작하여 검색할 최대 항목 수입니다(기본값은 100, 최대 100).
+
+
+## **`GET` /query/template**
+
+모든 쿼리 템플릿을 검색합니다.
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
모든 쿼리 템플릿을 불러옵니다.
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 15]
```diff
@@ -16,23 +16,34 @@ Execute ThanoSQL queries and receive a query log as a response.
     import json
 
     api_token = "Issued_API_TOKEN"
-    api_url="https://{your-engine-url}/api/v1/query/"
-    query="Query to request"
-    query_type="SQL query type" #psql or thanosql
+    api_url = "https://{your-engine-url}/api/v1/query/"
+    query_type = "SQL query type" #psql or thanosql
```
Comment (lazyduo) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
    query_type = "SQL query type" # psql or thanosql
```
ko도 바꿔주세요

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 16]
```diff
@@ -16,23 +16,34 @@ Execute ThanoSQL queries and receive a query log as a response.
...
+
     data = {
         'query_string': query, 'query_type': query_type
     }
 
-    r = requests.post(api_url, data=json.dumps(data), headers=header)
+    # For query with templates (pick 1 between this and above):
+    template_name = "Name of template to use"     # we don't need this line if we query by template_id
+    template_id = "ID number of template to use"  # we don't need this line if we query by template_name
+    parameters = "Mapping of parameters to fill in the template"
```
Comment (lazyduo) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
    parameters = "Mapping of parameters to fill in the template" # Dictionary
```
string이 아니라서 명시해줘야 할 것 같습니다. (ko도 마찬가지)

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 17]
```diff
@@ -80,20 +96,65 @@ ThanoSQL 쿼리를 실행하고 쿼리 로그를 응답으로 받습니다.
...
+    ```python
+    import requests
+    import json
+
+    api_token = "발급받은_API_TOKEN"
+    base_url = "https://{your-engine-url}/api/v1/query/template"
+    offset = {오프셋}
+    limit = {리밋}
+    search = "검색 키워드"
+    order_by = "Order by"
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
    order_by = "정렬 기준"
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 18]
```diff
@@ -108,12 +169,196 @@ ThanoSQL 쿼리를 실행하고 쿼리 로그를 응답으로 받습니다.
...
+    r.json()
+    ```
+
+=== "cURL"
+
+    ```shell
+    curl -X 'POST' \
+      'https://{your-engine-url}/api/v1/query/template' \
+      -H 'accept: application/json' \
+      -H 'Authorization: Bearer Issued_API_TOKEN' \
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
      -H 'Authorization: Bearer 발급받은_API_TOKEN' \
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 19]
```diff
@@ -108,12 +169,196 @@ ThanoSQL 쿼리를 실행하고 쿼리 로그를 응답으로 받습니다.
...
+    curl -X 'GET' \
+      'https://{your-engine-url}/api/v1/query/template/{template_name}' \
+      -H 'accept: application/json' \
+      -H 'Authorization: Bearer Issued_API_TOKEN'
+    ```
+
+
+## **`PUT` /query/template/{template_name}**
+
+특정 이름으로 쿼리 템플릿을 업데이트합니다.
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
특정 이름의 쿼리 템플릿을 업데이트합니다.
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 20]
```diff
@@ -108,12 +169,196 @@ ThanoSQL 쿼리를 실행하고 쿼리 로그를 응답으로 받습니다.
...
+    r.json()
+    ```
+
+=== "cURL"
+
+    ```shell
+    curl -X 'PUT' \
+      'https://{your-engine-url}/api/v1/query/template/{template_name}' \
+      -H 'accept: application/json' \
+      -H 'Authorization: Bearer Issued_API_TOKEN' \
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
      -H 'Authorization: Bearer 발급받은_API_TOKEN' \
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md [hunk 21]
```diff
@@ -77,11 +89,22 @@ title: Table Template APIs
...
       curl -X 'GET' \
-      'https://{your-engine-url}/api/v1/table_template/{table_template_name}' \
+      'https://{your-engine-url}/api/v1/table_template/{table_template_name}?version={version}' \
       -H 'accept: application/json' \
       -H 'Authorization: Bearer Issued_API_TOKEN'
     ```
 
+### __Response__
+
+이 API의 응답 구조는 다음과 같습니다:
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md]:
```suggestion
API의 응답 구조는 다음과 같습니다:
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md [hunk 22]
```diff
@@ -211,3 +234,39 @@ title: Table Template APIs
         "compatibility": "string"
         }'
     ```
+
+
+## **`DELETE` /table_template/{table_template_name}**
+
+특정 이름의 테이블 템플릿을 삭제하려면 이 메서드를 사용합니다. 필요한 경우 특정 버전만 삭제할 수도 있습니다.
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_table_templates.md]:
```suggestion
특정 이름의 테이블 템플릿을 삭제하려면 이 API를 호출합니다. 필요한 경우 특정 버전만 삭제할 수도 있습니다.
```

**Review by stevensjcho — COMMENTED** [review]
Make sure to translate the values in the cURL portion also!

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md
Comment (lazyduo) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md:1]: Let's split the API example into two methods. Mixing them all into one example makes it seem even more confusing.

```
## __`POST` /query__

### Query without Templates

### Query with Templates
```

**Review by lazyduo — COMMENTED** [review]
각 API의 설명글에 'Use this method' 라는 식의 설명은 빼고 'Retrieves all available query templates.', 'Creates ~~' 이런식으로 쓰는걸로 맞춰줍시다. 'Use this method'라고 써버리니 한글 번역에도 '이 메서드는'으로 표현되는게 보기 안좋네요.

참고 (https://cloud.google.com/bigquery/docs/reference/rest/v2/models/list)

**Review by stevensjcho — APPROVED** [review]
LGTM 👍

**Review by lazyduo — APPROVED** [review]
👍🏻

**Pull request merged**